### DISCUSSION TITLE
Feature request: Behave as Marquee when not dragging

### DISCUSSION THREAD

**Discussion by elron** [Valexr/Slidy#96] [Ideas]
Hey
I Imagine a way of having a marquee-like, with all the benefits of Slidy (drag / scroll)
Here's an explanation:
- If the user doesn't have any drag/scroll interaction, slidy will start to move pixel-by-pixel to the left/right, instead of freezing in place. (This is not the exact intention, but here's some inspiration: https://huanxinhu.github.io/react-drag-marquee/)
- When mouseDown (starts dragging), slidy will stop moving by itself, but react only to the user's drag.
- When mouseUp (stops dragging), slidy will start the marquee again.
- There should be multiple options like `marquee: {speed: 300, startAfter: 1000}` (`speed` = the speed of the marquee. if speed is negative number like -300, it will move the to other way. `startAfter` = when stops dragging, the time it takes to start the marquee again. if `startAfter` is 0, the marquee will start immediately after use stops dragging.)
- I believe it would be nice if the marquee animation will start gradually and increase over time, so it will seem smooth.

Is there a way to extend Slidy to behave like this?

I'd love to contribute to this if possible, but I need a starter.
[I've read here](https://github.com/Valexr/Slidy/tree/master/packages/svelte#external-controls) that position is bindable, but it doesn't seem to change the position of the slider itself, just the number.
Here's a repo of my test: https://svelte.dev/repl/f452f7f518454ad886977cd563b80fc3?version=3.50.0

Thanks

**Comment by Valexr** [comment]
Interesting idea… we thinking about @slidy/play plug-in with similar functionality… & in templates it’s realised for slideshow behaviour…

**Comment by elron** [comment]
Cool, sounds great. Can I help somehow?

by the way do you have any idea why Slidy doesn't respond to position change?
Try to press "move" button in this repo: https://svelte.dev/repl/082e40a606b5407cb4be74d6f0a938e1?version=3.50.0

If slidy could respond to the position change made by svelte, I can build the marquee functionality outside of the Slidy package, in the svelte environment.

**Comment by Valexr** [comment]
Now position it’s read only property & all navigation tied to the index…🤔 method `to(position)` had position, but removed now… Slidy was designed as carousel with index navigation, not as scrollable area… but… need thinking about free nav… now it’s possible with controls, but haven’t method or reactive position props…

help…🤔if you can Vue3… welcome make @slidy/vue… about marquee plug-in I need some time thinking about plug-in interface… & reactive position to 😉

**Comment by elron** [comment]
I see!
Unfortunately I don't use Vue3 but I can help with Sveltejs if you need.

About the plug-in & reactive position to, it would be great to make the position as a public API, because that way I can change it to my desire - and there are many creative things we can do.

- Simple way is to make it marquee, just like I suggested in the post.
- But another way is to make the speed & direction change depending on where the mouse cursor is. [Example](https://elad.elrons.co.il/) (press CTRL +D to view the debug mode, and then place the mouse in the blue parts- you would then see that the "marquee" changes its speed and direction).

What are your thoughts on this? any way to bring the position-to public?

**Comment by Valexr** [comment]
Reactive `position` [implemented](https://github.com/Valexr/Slidy/commit/ede73065fdd13bffc3281a5905735b30c810784f) - used only `snap: undefined` mode & will be released soon in @slidy/core@3.6.0  

About the `@slidy/play` plugin not sure what will be coming soon...

Stay tune ✌🏻

Speed controlled by `duration` option & you can use @slidy/core for yours custom svelte templates & working with true DOM nodes...

@slidy/svelte forwarding `play`, `pause` & `stop` events if you using `autoplay`

Maybe create @slidy/events plugin for forwarding needed event on target node...🤔

**Comment by elron** [comment]
Sounds amazing! Thank you so much for your effort

**Comment by Valexr** [comment]
`3.6.0` [released](https://github.com/Valexr/Slidy/releases/tag/3.6.0)
`3.7.0` with `@slidy/plugins` interface soon - plugin play [demo](https://slidy-core.surge.sh/)

**Comment by Valexr** [comment]
3.7.0 released https://github.com/Valexr/Slidy/releases/tag/3.7.0
i move this issue to disscussion
